apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Snippets/BigtableClientSnippets.cs

```diff
@@ -0,0 +1,367 @@
+            RowSet rows = RowSet.FromRowRanges(
+                RowRange.ClosedOpen("user00000", "user99999"));
+
+            // Only read cells from the "Address" column family and only take the most recent cell from each column.
+            RowFilter filter = RowFilters.Chain(
+                RowFilters.FamilyNameRegex("Address"),
+                RowFilters.CellsPerColumnLimit(1));
+
+            // Just read the first 10 rows.
+            long rowsLimit = 10;
+
+            // Make the request
+            BigtableClient.ReadRowsStream streamingResponse = client.ReadRows(
+                tableName,
+                rows,
+                filter,
+                rowsLimit,
+                CallSettings.FromCancellationToken(cancellationToken));
+
+            // Read streaming responses from server until complete
+            await streamingResponse.AsAsyncEnumerable().ForEachAsync(row =>
+            {
+                Console.WriteLine($"Row key: {row.Key.ToStringUtf8()}");
+                foreach (Family family in row.Families)
+                {
+                    Console.WriteLine($"  Family name: {family.Name}");
+                    foreach (Column column in family.Columns)
+                    {
+                        Console.WriteLine($"    Column qualifier: {column.Qualifier.ToStringUtf8()}");
+                        foreach (Cell cell in column.Cells)
+                        {
+                            Console.WriteLine($"      Cell value: {cell.Value.ToStringUtf8()}");
+                            Console.WriteLine($"      Timestamp: {cell.Version.ToDateTime()}");
+                            Console.WriteLine($"      Labels: {string.Join(", ", cell.Labels)}");
+                        }
+                    }
+                }
+            });
+            // End snippet
+        }
+
+        [Fact]
+        public async Task SampleRowKeys()
+        {
+            var cts = new CancellationTokenSource();
+            var cancellationToken = cts.Token;
+            var client = _fixture.TableClient;
+            var projectId = _fixture.TableName.ProjectId;
+            var instanceId = _fixture.TableName.InstanceId;
+            var tableId = _fixture.TableName.TableId;
+
+            // Snippet: SampleRowKeys(TableName,CallSettings)
+            BigtableClient.SampleRowKeysStream streamingResponse = client.SampleRowKeys(
+                new TableName(projectId, instanceId, tableId),
+                CallSettings.FromCancellationToken(cancellationToken));
+
+            // Get all streaming responses from server in a list so they can be easily iterated.
+            IList<SampleRowKeysResponse> rowKeySamples =
+                await streamingResponse.ToListAsync(cancellationToken);
+
+            long previousRowOffsetBytes = 0;
+            foreach (SampleRowKeysResponse rowKeySample in rowKeySamples)
+            {
+                long offsetDifference = rowKeySample.OffsetBytes - previousRowOffsetBytes;
```

Comment:
Is this guaranteed to always be positive?

Reply:
I would assume, but I'm not sure. Do you think it would make sense to add a `Math.Max(0, ...)` to be safe?

Reply:
I think we should probably find out what's guaranteed, and add a comment to that effect. If we use a Math.Max and there's no need to, that's bad... if we assume that it *will* always be positive and it's not, that's bad too.

Reply:
Sounds good. I'll ask internally, and update accordingly with a follow PR if that's ok with you.

Reply:
Ok, the response I got internally is that these "should be non-decreasing", so I'll add a comment.